packages/text-input/src/TextInput/TextInput.tsx

```diff
@@ -93,11 +65,10 @@ const TextInput = React.forwardRef<HTMLInputElement, TextInputProps>(
     }: TextInputProps,
     forwardRef: React.Ref<HTMLInputElement>,
   ) => {
-    const { darkMode, theme } = useDarkMode(darkModeProp);
+    const { darkMode } = useDarkMode(darkModeProp);
     const isControlled = typeof controlledValue === 'string';
     const [uncontrolledValue, setValue] = useState('');
     const value = isControlled ? controlledValue : uncontrolledValue;
-    const id = useIdAllocator({ prefix: 'textinput', id: propsId });
```

Comment:
Would this change make this a breaking change? If users were dependent on the value of this id always being `textinput-0` for example?